core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -308,6 +308,20 @@ sealed abstract class IO[+E, +A] { self =>
   final def ensuring(finalizer: IO[Nothing, Unit]): IO[E, A] =
     new IO.Ensuring(self, finalizer)
 
+  /**
+   * Executes the action on the specified `ExecutionContext` and then shifts back
+   * to the default one.
+   */
+  final def on(ec: ExecutionContext): IO[E, A] =
+    IO.shift(ec) *> self <* IO.shift
+
+  /**
+   * Forks an action that will be executed on the specified `ExecutionContext`
+   * and then shifts back to the default one.
+   */
+  final def forkOn(ec: ExecutionContext): IO[E, Fiber[E, A]] =
```

Comment:
I'm not sure this comment makes sense—because it's really only forking the `IO` _on_ the specified execution context. The main fiber which called `fork` never shifts off, so never needs to "shift back".